### PR TITLE
Break kubernetes/kubernetes deps in kubectl validation

### PR DESCRIPTION
**Pull request by pwittrock**
Switch from unversioned / internal libraries to published libraries for List and Scheme.

```release-note
NONE
```

### CODE REVIEW

**Comment by pwittrock** [comment]
cc @kubernetes/sig-cli-maintainers

File: pkg/kubectl/resource/result.go [hunk 1]
```diff
@@ -236,8 +242,12 @@ func AsVersionedObject(infos []*Info, forceList bool, version schema.GroupVersio
...
 		object = objects[0]
 	} else {
-		object = &api.List{Items: objects}
-		converted, err := TryConvert(api.Scheme, object, version, api.Registry.GroupOrDie(api.GroupName).GroupVersion)
+		e := []runtime.RawExtension{}
+		for _, o := range objects {
+			e = append(e, runtime.RawExtension{Object: o})
+		}
+		object = &v1.List{Items: e}
+		converted, err := TryConvert(scheme.Scheme, object, version, schema.GroupVersion{"", "v1"})
```
Comment (caesarxuchao) [inline pkg/kubectl/resource/result.go]: Is `object` supposed to be an internal versioned object? If so , scheme.Scheme won't be able to do the conversion. It only knows about external versions.

File: pkg/kubectl/resource/result.go [hunk 2]
```diff
@@ -268,7 +278,8 @@ func AsVersionedObjects(infos []*Info, version schema.GroupVersion, encoder runt
 		// objects that are not part of api.Scheme must be converted to JSON
 		// TODO: convert to map[string]interface{}, attach to runtime.Unknown?
 		if !version.Empty() {
-			if _, _, err := api.Scheme.ObjectKinds(info.Object); runtime.IsNotRegisteredError(err) {
+			var o runtime.Object = info.Object
+			if _, _, err := scheme.Scheme.ObjectKinds(o); runtime.IsNotRegisteredError(err) {
```
Comment (caesarxuchao) [inline pkg/kubectl/resource/result.go]: Same here. If `o` is of internal version, `scheme.Scheme` won't help.

Comment (liggitt) [reply pkg/kubectl/resource/result.go]: agree... AsVersionedObjects implies there is question as to whether the input objects are internal or of a different version... either implies conversion, which an external-only scheme won't be able to help with

**Comment by pwittrock** [comment]
@liggitt @caesarxuchao Thanks, that is what I was looking for.

**Pull request closed without merging**